2-projects/1-todo-list/script.js

```diff
@@ -15,11 +63,25 @@ populateTodoList(todos);
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
-  event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  let todoInput = document.getElementById("todoInput");
+  if (todoInput.value !== "") {
+    populateTodoList([{ task: todoInput.value, completed: false }]);
+    todoInput.value = "";
+  }
+  event.preventDefault();
 }
+document.querySelector("#add-todo").addEventListener("click", addNewTodo);
 
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
-function deleteAllCompletedTodos() {
-  // Write your code here...
+function deleteAllCompletedTodos(event) {
+  event.preventDefault();
+  let list = document.getElementById("todo-list");
+  for (const element of list.children) {
+    if (element.classList.contains("done")) {
+      list.removeChild(element);
```

Comment:
Look into using [this](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) instead of for loops. Then you can see the completed 'li' elements